firmware/src/hal/sgx/src/trusted/hmac_sha512.h

```diff
@@ -68,10 +68,23 @@
 /** Number of bytes a SHA-512 hash requires. */
 #define SHA512_HASH_LENGTH 64
 
-extern void hmac_sha512(uint8_t *out,
-                        const uint8_t *key,
-                        const unsigned int key_length,
-                        const uint8_t *text,
-                        const unsigned int text_length);
+/**
+ * @brief Calculate a 64 byte HMAC of an arbitrary message and key using
+ * SHA-512 as the hash function.
+ *
+ * @param out A byte array where the HMAC-SHA512 hash value will be written.
+ *            This must have space for SHA512_HASH_LENGTH bytes.
+ * @param key A byte array containing the key to use in the HMAC-SHA512
+ *            calculation. The key can be of any length.
+ * @param key_length The length, in bytes, of the key.
+ * @param text A byte array containing the message to use in the HMAC-SHA512
+ *             calculation. The message can be of any length.
+ * @param text_length The length, in bytes, of the message.
+ */
+void hmac_sha512(uint8_t *out,
```

Comment:
Probably a good time to change the return value to `bool` since there are paths that can fail with this implementation. We'd also need to account for this in `bip32.c`.